javascript/packages/fury/lib/meta/MetaString.ts

```diff
@@ -0,0 +1,148 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+class MetaString {
+  static LOWER_SPECIAL = 5;
+  static LOWER_UPPER_DIGIT_SPECIAL = 6;
+
+  // Encode function based on the specified bits per character
+  static encode(str: string, bitsPerChar: number): Uint8Array {
```

Comment:
I noticed that in Java, the Encoding flag is detected automatically:

```Java
  public MetaString encode(String input, Encoding[] encodings) {
    if (input.isEmpty()) {
      return new MetaString(input, Encoding.UTF_8, specialChar1, specialChar2, new byte[0]);
    }
    if (!StringSerializer.isLatin(input.toCharArray())) {
      return new MetaString(
          input,
          Encoding.UTF_8,
          specialChar1,
          specialChar2,
          input.getBytes(StandardCharsets.UTF_8));
    }
    Encoding encoding = computeEncoding(input, encodings);
    return encode(input, encoding);
  }
```
The bitsPerChar value is determined by the Encoding used. Perhaps, for consistency with Java's approach, we should handle this detail within our method. This could prevent confusion for users calling the encode method, who might otherwise be unsure about how to set the bitsPerChar argument.

Reply:
This  is the new  encode function which automatically detects , encoding and handles Emojis .  Please , are there any other test cases  ?
```js
  // Encode function with automatic encoding detection
  static encode(str: string): Uint8Array {
    const bitsPerChar = MetaString.detectEncoding(str);
    const totalBits = str.length * bitsPerChar + 1;
    const byteLength = Math.ceil(totalBits / 8);
    const bytes = new Uint8Array(byteLength);
    let currentBit = 1;

    for (const char of [...str]) { // Using spread operator to handle emojis correctly
      const value = bitsPerChar === MetaString.LOWER_SPECIAL
        ? MetaString.charToValueLowerSpecial(char)
        : MetaString.charToValueLowerUpperDigitSpecial(char);

      for (let i = bitsPerChar - 1; i >= 0; i--) {
        if ((value & (1 << i)) !== 0) {
          const bytePos = Math.floor(currentBit / 8);
          const bitPos = currentBit % 8;
          bytes[bytePos] |= (1 << (7 - bitPos));
        }
        currentBit++;
      }
    }

    const stripLastChar = bytes.length * 8 >= totalBits + bitsPerChar;
    if (stripLastChar) {
      bytes[0] |= 0x80;
    }
    return bytes;
  }
```

Reply:
can i use [TextEncoder
](https://developer.mozilla.org/en-US/docs/Web/API/TextEncoder) instead ?

Reply:
> can i use [TextEncoder ](https://developer.mozilla.org/en-US/docs/Web/API/TextEncoder) instead ?

Yes, the TextEncoder is implemented by both Node.js and browsers.